MorkoUnityProject/Assets/Scripts/LocalController.cs

```diff
@@ -13,31 +13,66 @@ public class LocalController
 		private Vector3 lastMousePosition = Input.mousePosition;
 		private LayerMask groundMask = 1 << 9;
 		
-		// mouseControl = True == control with mouse
-		// mouseControl = False == control with joystick
-		bool mouseControl = true;
+		// mouseRotate = True == rotate with mouse
+		// mouseRotate = False == rotate with joystick
+		// mouseMove = True == move with KB
+		// mouseMove = False == move with GP
+		bool mouseRotate = true;
+		bool keyboardMove = true;
 		
 		private PlayerSettings playerSettings;
-		private float movementSpeed;
+		private float walkSpeed;
 		private float sneakSpeed;
 		private float runSpeed;
+		private float sideMultiplier;
+		private float backwardMultiplier;
+		private float sideRunMultiplier;
+		private float backwardRunMultiplier;
+		private float accelerationWalk;
+		private float accelerateSneak;
+		private float accelerationRun;
+		private float decelerationWalk;
+		private float decelerationRun;
 		private bool isMorko = false;
 
+		private float currentMovementSpeed = 0f;
+		private float previousSpeed = 0f;
+		private Vector3 moveDirectionKeyboard;
+		private Vector3 moveDirectionGamepad;
+		private Vector3 lastPosition;
+		private Vector3 oldDirectionKeyboard = Vector3.zero;
+		private Vector3 oldDirectionGamepad = Vector3.zero;
+		private bool ran = false;
```

Comment:
What is this used for? Can we not  infer this value from currentMovementSpeed? If it is for animation, we will still probably use movement speed only. Remove if unnecessary.